packages/backend/src/config/FeatureFlags.ts

```diff
@@ -0,0 +1,85 @@
+export interface ResolvedFeatureFlag {
+  feature: string
+  enabled: boolean
+  used: boolean
+}
+
+export class FeatureFlags {
+  private readonly enabled = new Set<string>()
+  private readonly disabled = new Set<string>()
+  private readonly resolved = new Map<string, boolean>()
+  private readonly input: string
+
+  constructor(input: string) {
+    const items = input
+      .split(',')
+      .map((item) => item.trim())
+      .filter((item) => item.length > 0)
+    this.input = items.join(',')
+
+    for (const item of items) {
+      if (item.startsWith('!')) {
+        this.disabled.add(item.substring(1))
+      } else {
+        this.enabled.add(item)
+      }
+    }
+  }
+
+  with(input: string): FeatureFlags {
```

Comment:
this function name is too generic for me. I need to go to the implementation to understand what it's doing. Maybe `add`, `withFlag`, `append`, `push`. In context 

```typescript
const flags = getFlags()

// Current
flags.with('diffHistory')

// Proposed
flags.add('diffHistory')
flags.withFlag('diffHistory')
flags.append('diffHistory')
flags.push('diffHistory')
```